src/fairseq2/recipes/hf/datasets.py

```diff
@@ -0,0 +1,188 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+from typing import Any, Dict, Optional, Union, final
+
+from datasets import Dataset  # type: ignore
+
+from fairseq2.assets.card import AssetCard
+from fairseq2.data.data_pipeline import Collater, create_bucket_sizes, read_sequence
+from fairseq2.datasets.data_reader import DataPipelineReader
+from fairseq2.datasets.loader import DatasetLoader
+from fairseq2.gang import Gang
+
+HFBatch = Dict[str, Any]
+
+
+class HFDataset:
+    def __init__(self, name: str):
+        self.name = name
+    
+    def __repr__(self) -> str:
+        return self.name
+
+    def create_reader(
+        self,
+        dataset: Dataset,
+        gang: Gang,
+        bucket_size: Optional[int] = None,
```

Comment:
Please make parameters below `gang`, keyword-only by using the `<args>, *, <kwargs>` syntax of Python.